File: ios/Classes/model/flutter_default_custom/NLocale.swift
internal struct NLocale {
    let languageCode: String
    let countryCode: String?

    var localeStr: String {
        languageCode
    }

    static func fromMessageable(_ args: Any) -> NLocale? {
        let d = asDict(args)
        let languageCode: String = asString(d["languageCode"]!)
        if languageCode == "sys" {
            return nil
        }
        return NLocale(
                languageCode: languageCode,
                countryCode: castOrNull(d["countryCode"], caster: asString)
        )
    }
}

File: android/src/main/kotlin/dev/note11/flutter_naver_map/flutter_naver_map/model/base/NLocale.kt
package dev.note11.flutter_naver_map.flutter_naver_map.model.base

import dev.note11.flutter_naver_map.flutter_naver_map.converter.DefaultTypeConverter.asMap
import java.util.*

internal data class NLocale(
    val languageCode: String, val countryCode: String?,
) {

    fun toLocale(): Locale = if (countryCode == null) {
        Locale(languageCode)
    } else {
        Locale(languageCode, countryCode)
    }

    companion object {
        fun fromMessageable(rawMap: Any): NLocale? = rawMap.asMap().let { map ->
            val languageCode = map["languageCode"].toString()
            if (languageCode == "sys") null
            else NLocale(
                languageCode = languageCode,
                countryCode = map["countryCode"]?.toString()
            )
        }
    }
}

